content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+    {
+        public void GenFunction(T printvalue)
+        {
+            Console.WriteLine(printvalue)
+        }
+    }
+    public class program
+    {
+        public static void Main(string [] args)
+        {
+            Console.WriteLine("Printing integer value");
+            Generic<int> gen = new Generic<int>();
+            gen.GenFunction(10);
+            
+            Console.WriteLine("Printing string value");
+            Generic<string> genString = new Generic<string>();
+            gen.GenFunction("Hello world");
+            
+        }
+    }
+}
+```
+
+Note that _<T>_ is a placeholder for data types and replaced by datatype at compile time.
+
+### 2. Hashsets
+This is an implementation of a set. A set is like a big bag of entries with no particular order rather than a list. They are based on the discrete mathematic's idea of a 'set', which is the grouping of entries that can contain one, two, or many distinct items.  Sorting in a hash set is impossible because the entries have to be sorted. Use of a hash set is very much applicable where we have to store a large amount of data that doesn't have to be sorted eg storing the identity of cars passing through an entrance gate. A program example:
+
+```
+ using System;
+using System.Collections.Generic;
+  
+class MyHashSet {
+
+    static public void Main()
+    {
+  
+        // Creating HashSet
+     
+        HashSet<string> myhash1 = new HashSet<string>();
+  
+        // Add the elements in HashSet
+  
+        myhash1.Add("Ndung'u");
+        myhash1.Add("Elijah");
+
+        // Creating another HashSet
+     
+        HashSet<string> myhash2 = new HashSet<string>();
+        myhash2.Add("Section");
+        myhash2.Add("Git");
+   
+  
+        // Using ExceptWith method
+        myhash1.ExceptWith(myhash2);
+        foreach(var ele in myhash1)
+        {
+            Console.WriteLine(ele);
+        }
+    }
+}
+```
+
+We note that many operations available for the list are available for hash sets except that:
+
+* We CANNOT add an object at a specific position since the elements are not in any order.
+* We cannot replace an item for the same reason, although we can add and delete another.
+* Retrieving all the items is possible but the sequence is indeterminate.
+* It's meaningless to find what position an element is or rather to sort.
+
+### 3. Dictionaries
+Dictionaries are slightly different from lists and sets because instead of individual objects, they store pairs; a key and a value.
+The key contains the address to the pair. The value is the piece of information associated with the key.
+Dictionaries have a very wide market application for instance an address book in which keys are people's names, and the values are address, phone, email, etc.
+Duplicate keys are not permitted though values are. Also, a dictionary doesn't store items in order, but it can be sorted in order of keys to store a sorted dictionary. A simple program to illustrate:
```

Comment:
> A simple program to illustrate:

I would revise this to:

> Here is a simple program to illustrate:


Like I said before the other time you were introducing an example program, the transition to it wasn’t smooth. So I rephrased it to make it connect more with what was said last.